drivers/timer/stm32_lptim_timer.c

```diff
@@ -290,7 +290,7 @@ uint32_t z_clock_elapsed(void)
 	/* gives the value of LPTIM1 counter (ms)
 	 * since the previous 'announce'
 	 */
-	uint64_t ret = (lp_time * CONFIG_SYS_CLOCK_TICKS_PER_SEC) / LPTIM_CLOCK;
+	uint64_t ret = (uint64_t)((lp_time * CONFIG_SYS_CLOCK_TICKS_PER_SEC) / LPTIM_CLOCK);
 
```

Comment:
That doesn't look correct for me, the `lp_time * CONFIG_SYS_CLOCK_TICKS_PER_SEC` multiplication is still done as 32-bit. This should be written as:

`uint64_t ret = ((uint64_t)lp_time * CONFIG_SYS_CLOCK_TICKS_PER_SEC) / LPTIM_CLOCK;`

Reply:
I agree

Reply:
Thanks for the change, it looks correct now.